tests/cli/test_compose_tar_kickstart.sh

```diff
@@ -0,0 +1,158 @@
+        if ! rlCheckRpm httpd; then
+            dnf -y install httpd
+        fi
+        systemctl start httpd
+
+        ks_path="/var/www/html/ks-tar.cfg"
+        tmp_dir=$(mktemp -d /tmp/composer.XXXXX)
+        ssh_key_dir=$(mktemp -d /tmp/composer-ssh-keys.XXXXXX)
+
+        rlRun -t -c "ssh-keygen -t rsa -N '' -f $ssh_key_dir/id_rsa"
+        pub_key=$(cat $ssh_key_dir/id_rsa.pub)
+
+        bp_name="test-tar"
+        blueprint="$bp_name.toml"
+        cat > $blueprint << __EOF__
+name = "$bp_name"
+description = "tar image test"
+version = "0.0.1"
+modules = []
+
+[[packages]]
+name = "kernel"
+version = "*"
+
+[[packages]]
+name = "openssh-server"
+version = "*"
+
+[[groups]]
+name = "anaconda-tools"
+
+[[customizations.user]]
+name = "root"
+key = "$pub_key"
+
+__EOF__
+        $CLI blueprints push $blueprint
+        image_path="/var/www/html/root.tar.xz"
+
+        version=$(awk -F = '$1 == "VERSION_ID" { print $2 }' /etc/os-release | tr -d \")
+        arch=$(uname -m)
+        baseurl=$(curl "https://mirrors.fedoraproject.org/mirrorlist?repo=fedora-${version}&arch=${arch}" | \
+            grep -v "^#" | head -n 1)
+        curl --remote-name-all $baseurl/images/pxeboot/{vmlinuz,initrd.img}
+
+        fallocate -l 5G disk.img
+    rlPhaseEnd
+
+    rlPhaseStartTest "compose start"
+        rlAssertEquals "SELinux operates in enforcing mode" "$(getenforce)" "Enforcing"
+        uuid=$($CLI compose start $bp_name tar)
+        rlAssertEquals "exit code should be zero" $? 0
+
+        uuid=$(echo $uuid | cut -f 2 -d' ')
+    rlPhaseEnd
+
+    rlPhaseStartTest "compose finished"
+        if [ -n "$uuid" ]; then
+            until $CLI compose info $uuid | grep 'FINISHED\|FAILED'; do
+                sleep 10
+                rlLogInfo "Waiting for compose to finish ..."
+            done;
+        else
+            rlFail "Compose uuid is empty!"
+        fi
+
+        rlRun -t -c "$CLI compose image $uuid"
+        image="$uuid-root.tar.xz"
+    rlPhaseEnd
+
+    rlPhaseStartTest "Install tar image using kickstart liveimg command"
+        cat > $ks_path << __EOF__
+lang en_US.UTF-8
+timezone America/New_York
+keyboard us
+rootpw --lock
+sshkey --username root "$pub_key"
+bootloader --location=mbr
+zerombr
+clearpart --initlabel --all
+autopart
+# reboot is used together with --no-reboot qemu-kvm parameter for the qemu-kvm
+# process to exit after the installation is complete
+reboot
+
+liveimg --url http://10.0.2.2/root.tar.xz
+
+%post --log=/dev/console
+echo -e "curl http://10.0.2.2/image-$uuid-booted\n" >> /etc/rc.d/rc.local
+chmod +x /etc/rc.d/rc.local
+curl http://10.0.2.2/image-$uuid-installed
+%end
+__EOF__
+        mv $image $image_path
+        restorecon $image_path
+        $QEMU_BIN -m 2048 -drive file=disk.img,format=raw -nographic -kernel vmlinuz -initrd initrd.img \
+            -append "inst.ks=http://10.0.2.2/ks-tar.cfg inst.stage2=$baseurl console=ttyS0" --no-reboot &
+
+        timeout=60 # wait up to 60 minutes
```

Comment:
IMO 60 minutes is too much to wait, maybe something like 30 mins to wait for installation. Eitherway it takes around 20-25 min to build the image and the test will timeout when run in CI before the current timeout is exhausted.

Reply:
Actually, can you use `cmdline` in kickstart so we can see the progress in the log/terminal and also not start the qemu process in the background. Isn't it possible to block while the installation is running and then shutdown the VM/qemu ? From what I see this is exactly what you are doing (shutting down the VM, then boot it again) so there should be no need to start qemu in the background.

Reply:
OK, I'll lower the timeout (however in some cases that I observed, the installation took quite a long time (~45 minutes, possibly even more), so I hope that 30 minutes will be enough). I'll also change the installation type to cmdline.

However, as for the VM execution/teardown changes, I'm not sure if I understand what you suggest. The thing is that I have to start the machine first with direct kernel boot, perform the installation (and wait for the installation process to wholly finish and shutdown the machine). I can't just reboot, as it's not possible (or I couldn't find a suitable way) to boot from disk after the reboot. Technically I could overcome this by once booting from network (TFTP/iPXE) and then from the local drive, however I think this would be an unnecessary complication.

And if I ran the VM in the foreground, then I'd have to find some other way to manage for the timeout (I guess I'd have to run the timeout check in background, would this approach help us in any way?).

Reply:
Based on our yesterday's conversation, I performed the required changes for the qemu-kvm process to be blocking (and the timeout of the test will be managed by CI).